packages/core/src/common/env-variables/env-variables-protocol.ts

```diff
@@ -21,6 +21,11 @@ export interface EnvVariablesServer {
     getExecPath(): Promise<string>
     getVariables(): Promise<EnvVariable[]>
     getValue(key: string): Promise<EnvVariable | undefined>
+    getUserHomeFolder(): Promise<string>
+    getDataFolderName(): Promise<string>
+    getUserDataFolder(): Promise<string>
+    /** Windows specific. Returns system data folder of Theia. On other than Windows systems is the same as getUserDataFolder */
+    getAppDataFolder(): Promise<string>
```

Comment:
On Windows VSCode uses AppData folder for some info.

Reply:
Yes, it is for VS Code compatibility, but we should understand it is better. I don't think it is something Windows or VS Code specific.

For instance, Electron distinguishes between the app folder (`app.getPath('appData')`) and the user folder (`app.getPath('userData')`), see https://github.com/electron/electron/blob/master/docs/api/app.md#appgetpathname. If I understand correctly former is used for place where program stores its files and second one where a program stores user files. If it is true, we should put it like that and list examples of files in comments.

@thegecko @mcgordonite Maybe you can shed some light on this question :) I think this PR is important for you, it is cleanly separates how to configure user folder for the concrete product.

Reply:
> Yes, it is for VS Code compatibility

Can you please link the corresponding VS Code source? I couldn't find it with a text search for `getAppDataFolder` in the `vscode` repo.

Reply:
`userData` folder is simply the `appData` folder with the current application name at the end. It's not Windows specific.

e.g. on MacOS:

appData: ~/Library/Application Support
userData: ~/Library/Application Support/Mbed Studio (for example)

Reply:
@thegecko Do you know when it would be better to use appData instead of userData?

Reply:
@kittaakos in VS Code they seem to emulate Electron behaviour, but not using `app.getPath('userData')`: https://github.com/microsoft/vscode/blob/f3b191ab38fb9f1717ce5ce3396bb41204ffb399/src/paths.js#L18-L25

Reply:
I'm start thinking that we use `userData` here bogusly.  We probably should name it differently, user data it like where local storage is stored, not there settings or recent workspaces are stored.

Maybe we should call it `appHome` folder instead to avoid confusions with `userData` meaning from Electron? So there would be 4 folders:
- userHome: ~
- appHome = user home + home folder name:  ~/.theia
- appData: ~/Library/Application Support
- userData = appData + productName: ~/Library/Application Support/Mbed Studio (for example)

Does it make sense?

Reply:
@mmorhun I'm a bit confused, I could not find a single client making use of `getAppDataFolder`. Did I overlook something?

Reply:
> I could not find a single client making use of `getAppDataFolder`

That's why I have asked: why do we need it.

Reply:
let's keep it a naming as it add comments:
- `userDataFolder` controls a folder where user data like settings, recent workspaces and plubins are stored
- `appDataFolder` controls a folder where app data are stored, like plugin logs and storages

btw `PluginPathsServiceImpl.getLogsDirPath` and `PluginPathsServiceImpl.getWorkspaceStorageDirPath` are bogus, they should use `getAppDataFolder`, not `getUserDataFolder`

Also `getAppDataFolder` implementation should be aligned with https://github.com/microsoft/vscode/blob/f3b191ab38fb9f1717ce5ce3396bb41204ffb399/src/paths.js#L18-L25, except bits of VSCODE_DATA, APP_DATA is Electron variable and `USERPROFILE` is windows variable, using homedir always is bogus

Reply:
I've not seen it used, but you could imagine a company with multiple apps potentially sharing config data between them there?
I would be inclined to keep it and name it how an electron user would expect to see it.